src/pages/innsending.tsx

```diff
@@ -0,0 +1,54 @@
+import { GetServerSidePropsContext } from "next";
+import React from "react";
+import ErrorPage from "./_error";
+import { startSoknad } from "./api/quiz-api";
+import { getSession } from "@navikt/dp-auth/server";
+import { audienceDPSoknad } from "../api.utils";
+
+export async function getServerSideProps(context: GetServerSidePropsContext) {
+  if (process.env.NEXT_PUBLIC_LOCALHOST) {
+    return {
+      redirect: {
+        destination: `./uuid-innsending/`,
+        permanent: false,
+      },
+    };
+  }
+
+  const { token, apiToken } = await getSession(context);
+  if (!token || !apiToken) {
+    // TODO Redirect til hvilken login?
+    return {
+      redirect: {
+        destination: "/TODO-redoratoren-login",
+        permanent: false,
+      },
+    };
+  }
+
+  const onBehalfOfToken = await apiToken(audienceDPSoknad);
+  const soknadResponse = await startSoknad(onBehalfOfToken, "Innsending");
```

Comment:
Det er jo egentlig uuid-er som blir returnert fra det endepunktet så navnet er litt misvisende både for søknad og innsending.  Jeg og Nattaphong drodlet litt og kom frem til et forslag hvor vi splitter det i to forskjellige funksjoner selv om det kaller samme endepunkt i quiz. Det blir litt duplikat kode, men vi synes det blir mer lesbart.

```
export function createSoknadUuid(onBehalfOfToken: string) {
  return fetch(`${process.env.API_BASE_URL}/soknad`, {
    method: "Post",
    headers: {
      Authorization: `Bearer ${onBehalfOfToken}`,
    },
  });
}

export function createInnsendingUuid(onBehalfOfToken: string) {
  return fetch(`${process.env.API_BASE_URL}/soknad?prosesstype=Innsending`, {
    method: "Post",
    headers: {
      Authorization: `Bearer ${onBehalfOfToken}`,
    },
  });
}

```

Reply:
Jeg er enig, jeg endrer :+1:

Reply:
Vi har noen ganger snakket om at `dp-soknad` og `Søknad` egentlig burde vært `dialog`, for vi (kan) ha mange ulike dialoger, hvor søknad er en av de.